sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -71,7 +71,7 @@ trait ProgressReporter extends Logging {
   private var metricWarningLogged: Boolean = false
 
   /** Holds the most recent query progress updates.  Accesses must lock on the queue itself. */
-  private val progressBuffer = new mutable.Queue[StreamingQueryProgress]()
+  @transient private val progressBuffer = new mutable.Queue[StreamingQueryProgress]()
```

Comment:
mark `progressBuffer` `@transient` instead of changing it to serializable because it may contain a lot of progresses and we should try to avoid sending them to executors.